clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -2809,6 +2817,65 @@ private static void cleanupInstanceCustomizedStates(CommandLine cmd) {
     printObject(multiStoreTopicsResponse);
   }
 
+  private static void getRequestBasedMetadata(CommandLine cmd) throws JsonProcessingException {
+    String url = getRequiredArgument(cmd, Arg.URL);
+    String serverUrl = getRequiredArgument(cmd, Arg.SERVER_URL);
+    String storeName = getRequiredArgument(cmd, Arg.STORE);
+    ClientConfig clientConfig = ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(serverUrl);
+    if (clientConfig.isHttps()) {
+      if (!sslFactory.isPresent()) {
+        throw new VeniceException("HTTPS url requires admin tool to be executed with cert");
+      }
+      clientConfig.setSslFactory(sslFactory.get());
+    }
+    TransportClient transportClient = null;
+    try {
+      transportClient = ClientFactory.getTransportClient(clientConfig);
+      ControllerClient metadataResponseSchemaCC = ControllerClientFactory.discoverAndConstructControllerClient(
+          AvroProtocolDefinition.SERVER_METADATA_RESPONSE.getSystemStoreName(),
+          url,
+          sslFactory,
+          1);
+      getAndPrintRequestBasedMetadata(transportClient, metadataResponseSchemaCC, serverUrl, storeName);
+    } finally {
+      Utils.closeQuietlyWithErrorLogged(transportClient);
+    }
+  }
+
+  static void getAndPrintRequestBasedMetadata(
+      TransportClient transportClient,
+      ControllerClient controllerClient,
+      String serverUrl,
+      String storeName) throws JsonProcessingException {
+    String requestBasedMetadataURL = QueryAction.METADATA.toString().toLowerCase() + "/" + storeName;
+    byte[] body;
+    int writerSchemaId;
+    try {
+      TransportClientResponse transportClientResponse = transportClient.get(requestBasedMetadataURL).get();
+      writerSchemaId = transportClientResponse.getSchemaId();
+      body = transportClientResponse.getBody();
+    } catch (Exception e) {
+      throw new VeniceException(
+          "Encountered exception while trying to send metadata request to: " + serverUrl + "/"
+              + requestBasedMetadataURL,
+          e);
+    }
+    SchemaResponse schemaResponse = controllerClient
+        .getValueSchema(AvroProtocolDefinition.SERVER_METADATA_RESPONSE.getSystemStoreName(), writerSchemaId);
```

Comment:
One optimization is to only query controller if the schema is not available locally. It's just a nit, but can be helpful if we end up scripting this thing